src/Bridge/Doctrine/Orm/Extension/OrderExtension.php

```diff
@@ -47,10 +47,22 @@ public function applyToCollection(QueryBuilder $queryBuilder, QueryNameGenerator
             if (null !== $defaultOrder) {
                 foreach ($defaultOrder as $field => $order) {
                     if (is_int($field)) {
+                        // Default direction
                         $field = $order;
                         $order = 'ASC';
                     }
-                    $queryBuilder->addOrderBy('o.'.$field, $order);
+                    if (false === ($pos = strpos($field, '.'))) {
+                        // Configure default filter with property
+                        $field = 'o.'.$field;
+                    } else {
+                        // Configure default filter with association
+                        // Ensure query join is done on association
+                        $associationName = substr($field, 0, $pos);
+                        if (!in_array($associationName, $queryBuilder->getRootAliases(), true)) {
+                            $queryBuilder->join('o.'.$associationName, $associationName);
```

Comment:
No conflicts with [src/Bridge/Doctrine/Orm/Extension/FilterEagerLoadingExtension.php#L117-L124](https://github.com/api-platform/core/blob/babe7bbb3a4ca5c40a111cbe41ce81f309677b5d/src/Bridge/Doctrine/Orm/Extension/FilterEagerLoadingExtension.php#L117-L124) here (extensions priorities)?

Reply:
The helper (as long as it is marked `@internal`) looks good to me. The QueryBuilder is provided by Doctrine, it's not really wanted to modify it.

Reply:
I'd keep static helpers instead of extending the QueryBuilder.